src/MSALWrapper/AuthFlow/IntegratedWindowsAuthentication.cs

```diff
@@ -0,0 +1,170 @@
+                try
+                {
+                    try
+                    {
+                        var tokenResult = await TaskExecutor.CompleteWithin(
+                            this.logger,
+                            this.silentAuthTimeout,
+                            "Get Token Silent",
+                            (cancellationToken) => this.pcaWrapper.GetTokenSilentAsync(this.scopes, account, cancellationToken),
+                            this.errors)
+                            .ConfigureAwait(false);
+                        tokenResult.SetAuthenticationType(AuthType.Silent);
+
+                        return new AuthFlowResult(tokenResult, this.errors);
+                    }
+                    catch (MsalUiRequiredException ex)
+                    {
+                        this.errors.Add(ex);
+                        this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
+                    }
+                }
+                catch (MsalServiceException ex)
+                {
+                    this.logger.LogWarning($"MSAL Service Exception! (Not expected)\n{ex.Message}");
+                    this.errors.Add(ex);
+                }
+                catch (MsalClientException ex)
+                {
+                    this.logger.LogWarning($"Msal Client Exception! (Not expected)\n{ex.Message}");
+                    this.errors.Add(ex);
+                }
+                catch (NullReferenceException ex)
+                {
+                    this.logger.LogWarning($"Msal unexpected null reference! (Not Expected)\n{ex.Message}");
+                    this.errors.Add(ex);
+                }
+            }
+            else
+            {
+                try
+                {
+                    var tokenResult = await TaskExecutor.CompleteWithin(
+                                    this.logger,
+                                    this.interactiveWindowsAuthTimeout,
+                                    "Get Token Interactive Windows Authentication",
+                                    (cancellationToken) => this.pcaWrapper.GetTokenIntegratedWindowsAuthenticationAsync(this.scopes, cancellationToken),
+                                    this.errors)
+                                    .ConfigureAwait(false);
+                    tokenResult.SetAuthenticationType(AuthType.IntegratedWindowsAuthenticationFlow);
+
+                    return new AuthFlowResult(tokenResult, this.errors);
+                }
+                catch (MsalUiRequiredException ex) when (
+                             ex.Classification == UiRequiredExceptionClassification.BasicAction
+                          && ex.Message.StartsWith("AADSTS50076", StringComparison.OrdinalIgnoreCase))
+                {
+                    this.errors.Add(ex);
+                    this.logger.LogDebug($"IWA failed, 2FA is required.\n{ex.Message}");

```

Comment:
Since we gather this only for our telemetry, do you have any specific suggestions?

Reply:
Perhaps we can use `LogWarning` here to note that IWA failed with 2FA required and tell them, IWA can pass this requirement if you log into Windows with either a Smart Card or Windows Hello. This way it's shown to the user, as we go onto to do some interactive auth. This way they know more about why they are getting prompted and what to do.

Reply:
Changing LogDebug to LogWarning so that users are get helpful information before we try to run an interactive prompt. This way they know how to avoid interactive prompts next time.